user_modules/shared_testcases/testCasesForExternalUCS.lua

```diff
@@ -0,0 +1,179 @@
+	          :Do(
+	            function(_, d)
+	              test.hmiConnection:SendResponse(d.id, "BasicCommunication.SystemRequest", "SUCCESS", { })
+	              test.hmiConnection:SendNotification("SDL.OnReceivedPolicyUpdate",
+	                { policyfile = policy_file_path .. "/" .. policy_file_name })
+	            end)
+	            test["mobileSession"..id]:ExpectResponse(corIdSystemRequest, { success = true, resultCode = "SUCCESS"})
+          end)
+      end)
+    os.remove(ptu_file_name)
+  end
+
+	function utils.startSession(test, id)
+    test["mobileSession"..id] = mobile_session.MobileSession(test, test.mobileConnection)
+    test["mobileSession"..id]:StartService(7)
+  end
+
+	function utils.registerApp(test, id)
+		local RAIParams = config["application"..id].registerAppInterfaceParams
+    local corId = test["mobileSession"..id]:SendRPC("RegisterAppInterface", RAIParams)
+    EXPECT_HMINOTIFICATION("BasicCommunication.OnAppRegistered",
+      { application = { appName = RAIParams.appName } })
+    :Do(
+      function(_, d)
+        utils.HMIAppIds[RAIParams.appID] = d.params.application.appID
+      end)
+    test["mobileSession"..id]:ExpectResponse(corId, { success = true, resultCode = "SUCCESS" })
+    :Do(
+      function()
+        test["mobileSession"..id]:ExpectNotification("OnHMIStatus",
+          { hmiLevel = "NONE", audioStreamingState = "NOT_AUDIBLE", systemContext = "MAIN" })
+        test["mobileSession"..id]:ExpectNotification("OnPermissionsChange")
+      end)
+	end
+
+	function utils.activateApp(test, id)
+		local appId = config["application"..id].registerAppInterfaceParams.appID
+		local reqId = test.hmiConnection:SendRequest("SDL.ActivateApp", { appID = utils.HMIAppIds[appId] })
+    EXPECT_HMIRESPONSE(reqId)
+    :Do(
+      function(_, d1)
+        if d1.result.isSDLAllowed ~= true then
+          local reqId2 = test.hmiConnection:SendRequest("SDL.GetUserFriendlyMessage",
+            { language = "EN-US", messageCodes = { "DataConsent" } })
+          EXPECT_HMIRESPONSE(reqId2)
+          :Do(
+            function()
+              test.hmiConnection:SendNotification("SDL.OnAllowSDLFunctionality",
+                { allowed = true, source = "GUI", device = { id = config.deviceMAC, name = "127.0.0.1" } })
+              EXPECT_HMICALL("BasicCommunication.ActivateApp")
+              :Do(
+                function(_, d2)
+                  test.hmiConnection:SendResponse(d2.id,"BasicCommunication.ActivateApp", "SUCCESS", { })
+                  test["mobileSession"..id]:ExpectNotification("OnHMIStatus",
+                    { hmiLevel = "FULL", audioStreamingState = "AUDIBLE", systemContext = "MAIN" })
+                end)
+            end)
+        end
+      end)
+	end
+
+	function utils.policyUpdate(test, id, status, updateFunc)
+		EXPECT_HMICALL("BasicCommunication.PolicyUpdate")
+    :Do(
+      function(e, d)
+        if e.occurences == 1 then -- workaround due to issue
```

Comment:
@istoimenova, I've checked in O-S - issue is not in place, so function will be updated